hubblestack_nova/modules/cve_scan_v2.py

```diff
@@ -118,97 +112,113 @@ def audit(data_list, tags, verbose=False):
-            for local_version in local_pkgs[local_pkg]:
-                # There can be multiple cve announcements for a single package, check against all
-                for affected_obj in affected_pkgs[local_pkg]:
-                    affected_version = affected_obj.pkg_version
-                    if _is_vulnerable(local_version, affected_version, affected_obj.operator):
-                        # If the local pkg hasn't been found as vulnerable yet, vulnerable is None
-                        if not vulnerable:
-                            affected_obj.oudated_version = local_version
-                            vulnerable = affected_obj
-                        # If local_pkg has already been marked affected, vulnerable is set. We
-                        #   want to report the most recent cve, so check if the new affected_pkg
-                        #   version number is greater than the previously found vulnerability.
-                        else:
-                            if _is_vulnerable(vulnerable.pkg_version, affected_version, 'lt'):
-                                # If affected_obj version is > vulnerable, reassign vulnerable
+    for url, cache, cached_json, cached_zip, min_score in endpoints:
+        if cache: # Valid cached file
+            master_json = cache
+        else: # Query the url for cve's
+            if url.startswith('http://') or url.startswith('https://'):
+                if 'vulners.com' in url:
+                    # Vulners api can only handles http:// requests from request.get
+                    if url.startswith('https'):
+                        url.replace('https', 'http', 1)
+                    # Format the url for the request based on operating system.
+                    if url.endswith('/'):
+                        url = url[:-1]
+                    url_final = '%s/api/v3/archive/distributive/?os=%s&version=%s' \
+                                                                % (url, os_name, os_version)
+                    cve_query = requests.get(url_final)
+                    # Confirm that the request was valid.
+                    if cve_query.status_code != 200:
+                        raise Exception('Vulners requests was not successful. Check the url.')
+                    # Save vulners zip attachment in cache location and extract json
+                    try:
+                        with open(cached_zip, 'w') as zip_attachment:
+                            zip_attachment.write(cve_query.content)
+                        zip_file = ZipFile(cached_zip)
+                        zip_file.extractall(os.path.dirname(cached_zip))
+                        os.remove(cached_zip)
+                        extracted_json = os.path.join(__opts__['cachedir'],
+                                                      'cve_scan_cache',
+                                                      '%s_%s.json' % (os_name, os_version))
```

Comment:
I think that we need `os_version.replace('.', '')`, because in the case of Ubuntu-16.04, the zip file that the vulners api returns to us is `ubuntu_1604.zip` and extracted to `ubuntu_1604.json`. Or does `os.path.join(...)` handle omitting the period for us?